src/HotChocolate/ApolloFederation/src/ApolloFederation/RequiresAttribute.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Reflection;
+using HotChocolate.ApolloFederation.Extensions;
+using HotChocolate.Types;
+using HotChocolate.Types.Descriptors;
+
+namespace HotChocolate.ApolloFederation
+{
+    [AttributeUsage(AttributeTargets.Property)]
+    public class RequiresAttribute : DescriptorAttribute
+    {
+        public RequiresAttribute(string fieldSet)
+        {
+            FieldSet = fieldSet;
+        }
+
+        public string FieldSet { get; }
+
+        protected override void TryConfigure(
+            IDescriptorContext context,
+            IDescriptor descriptor,
+            ICustomAttributeProvider element)
+        {
+            if (descriptor is IObjectFieldDescriptor ofd)
+            {
```

Comment:
check for empty fieldset

Reply:
I added a check with a todo helper comment